sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -350,7 +350,7 @@ object FunctionRegistry {
     expression[StringTrimLeft]("ltrim"),
     expression[JsonTuple]("json_tuple"),
     expression[ParseUrl]("parse_url"),
-    expression[StringLocate]("position"),
+    expression[StringLocate]("position", true),
```

Comment:
This case, you should also manually handle the aliases at `override def prettyName: String = "locate"`

Reply:
OK. I will open another PR to manually handle the aliases. This PR is used to add more powerful guarantees so that aliases outputed by built-in functions are correct.

Reply:
There already exists
`expression[StringLocate]("locate"),`
and `StringLocate` has the override method.
`override def prettyName: String = "locate"`

Reply:
I think we can just fix one line here. See https://github.com/apache/spark/pull/27901/files#diff-b83497f7bc11578a0b63a814a2a30f48R103-R104